modules/middleware/src/aggregations.js

```diff
@@ -604,17 +606,17 @@ export default class AggregationProcessor {
           break;
 
         case 'filtered':
-          // ORIGINAL: v = { k: innerValue for k, innerValue in v.items() if re.match(r"{}(:(stats|histogram))?".format(field), k) is not None}
-          const filteredEntries = Object.entries(v).filter(item => {
-            const key = item[0];
-            return [`${field}:stats`, `${field}:histogram`].includes(key);
-          });
-
-          const innerValues = filteredEntries.map(item => item[1]);
-          v = {
-            [k]: innerValues,
-          };
-          break;
+          // original python code
+          /*
+          b = { k: innerValue for k, innerValue in v.items() if re.match(r"{}(:(stats|histogram))?".format(field), k) is not None}
+          */
+          v = Object.entries(v).reduce((acc, [k, v]) => {
```

Comment:
could we name the value in side the reduce something else? Was a little confusing as `v` is also used outside

Reply:
confusing you say! yes this was horribly confusing and I kept it for some reason. will fix